src/pages/search/component.js

```diff
@@ -0,0 +1,81 @@
+import React, { useState, useEffect } from 'react'
+import style from './style.module.scss'
+// ROUTER
+import { useNavigate, useLocation } from 'react-router-dom'
+
+const Search = () => {
+    // STATE
+    const [movie, setMovie] = useState(null)
+	const [error, setError] = useState(null)
+	const [isPending, setIsPending] = useState(false)
+    const navigate = useNavigate()
+
+    // extract the search query parameters
+    const queryString = useLocation().search;
+    const queryParams = new URLSearchParams(queryString)
+    const query = queryParams.get('query')
+
+    // useEFFECT
+    const fetchMovie = async () => {
+        setIsPending(true)
+
+		try{
+			const response = await fetch(`http://localhost:4000/movies?search=${query}&searchBy=title`)
+            console.log(response);
+			if(!response){
+				throw new Error(response.statusText)
+			}
+
+            if(!response.data){
+                throw new Error('No such movie found')
+            }
+
+			const json = await response.json()
+			setIsPending(false)
```

Comment:
same for error and pending states of the request - it all can be done in store
if you choose structure of your store like this:
```
movie: {
  data:...,
  isLoading:...,
  isError:...,
}
```
you can then manipulate this statuses same way you do it here, but with Redux